### PR TITLE
Split ordering cases for txs

### PR DESCRIPTION
**Pull request by nikitosing**
Continuation of https://github.com/blockscout/blockscout/pull/6502

## Checklist for your Pull Request (PR)

  - [x] I added an entry to `CHANGELOG.md` with this PR
  - [ ] If I added new functionality, I added tests covering it.
  - [ ] If I fixed a bug, I added a regression test to prevent the bug from silently reappearing again.
  - [ ] I checked whether I should update the docs and did so by submitting a PR to https://github.com/blockscout/docs
  - [ ] If I added/changed/removed ENV var, I submitted a PR to https://github.com/blockscout/docs to update the list of env vars at https://github.com/blockscout/docs/blob/master/for-developers/information-and-settings/env-variables.md and I updated the version to `master` in the Version column. Changes will be reflected in this table: https://docs.blockscout.com/for-developers/information-and-settings/env-variables. 
  - [ ] If I add new indices into DB, I checked, that they are not redundant with PGHero or other tools

### CODE REVIEW

**Comment by vbaranov** [comment]
@nikitosing could you please indicate which query(ies) exactly, these indexes will be used for?

**Comment by nikitosing** [comment]
@vbaranov for this query https://github.com/blockscout/blockscout/blob/366c1de9599c797f5d3409c1d5515aadef032b68/apps/explorer/lib/explorer/chain.ex#L4386
in such cases:
https://github.com/blockscout/blockscout/blob/366c1de9599c797f5d3409c1d5515aadef032b68/apps/explorer/lib/explorer/chain.ex#L4393-L4411

File: apps/explorer/priv/repo/migrations/20221126103223_add_txs_indexes.exs [hunk 1]
```diff
@@ -0,0 +1,46 @@
+defmodule Explorer.Repo.Migrations.AddTxsIndexes do
```
Comment (vbaranov) [inline apps/explorer/priv/repo/migrations/20221126103223_add_txs_indexes.exs:1]: @nikitosing how about current indexes for the version of the query before changes?
```
    "transactions_created_contract_address_hash_recent_collated_inde" btree (created_contract_address_hash, block_number DESC, index DESC, ha
sh)
    "transactions_from_address_hash_recent_collated_index" btree (from_address_hash, block_number DESC, index DESC, hash)
    "transactions_to_address_hash_recent_collated_index" btree (to_address_hash, block_number DESC, index DESC, hash)
```
?
Is any rationale to keep them?

Comment (nikitosing) [reply apps/explorer/priv/repo/migrations/20221126103223_add_txs_indexes.exs:1]: We use new ordering only in 2 cases:
- this is first page loading of txs for address
- address has more than 50 pending transactions and we loading the next page

In all other cases use old order_by: `block_number DESC, index DESC`

**Review by vbaranov — APPROVED** [review]

**Pull request merged**